python/tank/templatekey.py

```diff
@@ -292,31 +304,164 @@ def __init__(self,
                                          abstract=abstract,
                                          length=length)
 
-        if not(format_spec is None or isinstance(format_spec, basestring)):
-            msg = "Format_spec for TemplateKey %s is not of type string: %s"
+    def _init_format_spec(self, name, format_spec):
```

Comment:
I think it's good practice (we do it in a lot of places) to initialise member variables in the contructor. So I would love to see something like:

```
    self._padding_char = None
    self._minimum_width = None   
    self.format_spec = None
    self._init_format_spec(name, format_spec)
```